src/core/algorithm/plan.h

```diff
@@ -15,6 +15,22 @@ class RefPlan;
 
 class Plan {
 public:
+    // Event which may happen during the iteration.
+    // Extend Type if you want to use more types of events.
+    struct Event {
+        enum Type {
+            FALL_OJAMA_ROWS,
+        };
+
+        // This figures when this event will happen. 0 figures the frameId when a API is called.
+        int frames;
+
+        // Meaning of this variable depends on which type of event.
+        int value;
+
+        Type type;
+    };
```

Comment:
I'm OK with another patch, but probably it's good we have a utility method to make FALL_OJAMA_ROWS type event. (if so, we can write a comment about what `value` means for FALL_OJAMA_ROWS type in that method.)